### PR TITLE
Unit Test: Construction of DataFrame works with empty list but not empty array

### LINKED ISSUE #46822: BUG: Construction of DataFrame works with empty list but not empty array
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
df = pandas.DataFrame(data=[], columns=['a','b']) # works fine
df = pandas.DataFrame(data=numpy.array([]), columns=['a','b'])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.9/site-packages/pandas/core/frame.py", line 694, in __init__
    mgr = ndarray_to_mgr(
  File "/usr/local/lib/python3.9/site-packages/pandas/core/internals/construction.py", line 351, in ndarray_to_mgr
    _check_values_indices_shape_match(values, index, columns)
  File "/usr/local/lib/python3.9/site-packages/pandas/core/internals/construction.py", line 418, in _check_values_indices_shape_match
    raise ValueError("Empty data passed with indices specified.")
ValueError: Empty data passed with indices specified.
```

### Issue Description

Constructing an empty dataframe with an empty list works fine, but giving an empty numpy array, which intuitively should behave the same as with an empty list, results in a mysterious `ValueError: Empty data passed with indices specified.`

### Expected Behavior

Should create an empty dataframe with whatever columns listed.

### Installed Versions

INSTALLED VERSIONS
------------------
commit           : 4bfe3d07b4858144c219b9346329027024102ab6
python           : 3.9.7.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 21.4.0
Version          : Darwin Kernel Version 21.4.0: Mon Feb 21 20:34:37 PST 2022; root:xnu-8020.101.4~2/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.4.2
numpy            : 1.21.4
pytz             : 2021.1
dateutil         : 2.8.2
pip              : 21.3.1
setuptools       : 57.4.0
Cython           : 0.29.24
pytest           : 6.2.5
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 3.0.1
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.1
IPython          : 7.25.0
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
markupsafe       : 2.0.1
matplotlib       : 3.4.3
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : 0.15.0
pyarrow          : 5.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.7.1
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : 0.8.9
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None

### PR DESCRIPTION
**Pull request by weikhor**
- [ ] closes #46822 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by phofl — COMMENTED** [review]
Do we really want that?

**Review by jreback — CHANGES_REQUESTED** [review]
this is as intended, so i don't think we want to change.

**Review by mroeschke — COMMENTED** [review]
@weikhor instead could you just write a unit test `pandas.DataFrame(data=numpy.array([]), columns=['a','b'])` raises an exception cementing this behavior?

File: pandas/tests/frame/test_constructors.py [hunk 1]
```diff
@@ -3084,3 +3084,9 @@ def test_tzaware_data_tznaive_dtype(self, constructor):
 
         assert np.all(result.dtypes == "M8[ns]")
         assert np.all(result == ts_naive)
+
+    def test_construction_empty_dataframe(self):
```
Comment (mroeschke) [inline pandas/tests/frame/test_constructors.py:3088]: Could you make this test name a bit more specific like `test_construction_empty_array_multi_column_raises`

**Review by mroeschke — APPROVED** [review]
Thanks LGTM. Needs @jreback approval to merge

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @weikhor

**Pull request merged**